content/root/best-practice-guide.textile

```diff
@@ -19,28 +19,57 @@ h2(#ably-101). Ably 101
+
+If you prefer to use a single auth strategy across all third-party services, you can use "JWT Authentication":/core-features/authentication#ably-jwt-process as part of Token authentication.
+
+h3(#using-api-key-client-side). Using the API key on the client side
+
+Ably's "Basic authentication":/core-features/authentication#basic-authentication method is the easiest way to authenticate your clients on the Ably platform, but, this will directly expose the API key in the source code. Hence, when using it on the client side, any entity that interacts with the client could view your API key and potentially misuse it. So, if you can run an auth server, we recommend using "Token authentication":/core-features/authentication#token-authentication instead. With this method, your auth server is expected to use the API key to raise a token request from Ably, and the token can then be used by your end clients in the authentication step, eliminating the chance of misusing an exposed API key. "Read more":/core-features/authentication#selecting-auth about selecting an authentication mechanism.
+
+h3(#ensuring-tokens-are-auto-renewable). Ensuring tokens are auto-renewable
+
+Tokens are short-lived i.e. they expire after a given time. It is almost never correct to provide a token to a client library without ensuring the client library has a means to get a new token automatically when it needs one. The Ably client library SDKs provide two mechanisms in the "ClientOptions":/realtime/authentication#token-authentication so that they can obtain tokens automatically using either "@authUrl@":/realtime/authentication#token-upgrading or the "@authCallback@":/realtime/authentication#token-upgrading.
 
 h3(#rest-or-realtime). Realtime vs REST
 
-Ably provides two types of client libraries. "Realtime":/realtime libraries maintain a persistent connection to Ably and can be considered stateful, whereas "REST":/rest libraries issue HTTP requests to the Ably platform and are stateless.
+Ably Client Library SDKs may implement two interfaces: realtime and REST. The "Realtime":/realtime  interface maintains a persistent connection to Ably and can be considered stateful, whereas the "REST":/rest interface issues HTTP requests to the Ably platform and is stateless. The REST interface is therefore much lighter weight as it does not need the complexity of maintaining state. The REST interface can also only publish, but not subscribe to messages.
+
+Depending on your use case, you may only need to use the REST interface on your server, as it may only need to authenticate with Ably. You typically use the realtime interface on your client.
 
-h3(#auth). Authentication and Encryption
+The PHP and Python SDKs only have a REST interface. While the Ruby SDK implements both realtime and REST interfaces, there is also a Ruby SDK available that implements only the REST interface. 
```

Comment:
Would be really nice to have a matrix either here or linked to elsewhere, that shows which interfaces the various different client libs support?

Reply:
I've added a link to the download page which has a complete list.